### PR TITLE
fix: do not use fixed size buffer

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Using fixed size buffer was causing message truncate during read.

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt [hunk 1]
```diff
@@ -13,6 +16,8 @@ import java.util.*
 // Server implementation based on https://github.com/teocci/NioSocketCodeSample/tree/master
 class CopilotServerServiceImpl(private val project: Project): CopilotServerService {
 
+    private val LOG: Logger = Logger.getInstance(CopilotPluginUtil::class.java)
```
Comment (czp13) [inline src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:19]: Should not this be?:
```suggestion
    private val LOG: Logger = Logger.getInstance(CopilotServerServiceImpl::class.java)
```

Seems like pasted and the class name was not rewritten, not a big thingie, log entry will contain the other class name not this one.

Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:19]: No, by purpose I am using same logger to easily allow disabling it by single rule

Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:19]: Maybe it should more generic name, like just "Copilot"

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt [hunk 2]
```diff
@@ -109,28 +114,26 @@ class CopilotServerServiceImpl(private val project: Project): CopilotServerServi
...
+                    key.cancel()
+                    break
+                }
+                baos.write(readBuffer.array(), 0, read)
+            }
         } catch (e: IOException) {
-            e.printStackTrace()
+            LOG.error(e)
             key.cancel()
             channel.close()
```
Comment (czp13) [inline src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:132]: Shouldn't the channel also close when we finish reading, not just when we catch an error? The previous implementation closed it in the if(read == -1) condition as well.

Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:132]: Closing is done by client, but indeed, server should also close it

File: src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt (hunk 1 above)
Comment (czp13) [reply src/main/kotlin/com/vaadin/plugin/copilot/service/CopilotServerServiceImpl.kt:19]: Ah, okay, Copilot or Copilot-Plugin, okay, I see! Then we can leave it as it is.

**Review by czp13 — APPROVED** [review]

**Review by czp13 — APPROVED** [review]

**Pull request merged**